packages/fxa-customs-server/lib/email_record.js

```diff
@@ -197,6 +199,36 @@ module.exports = function(limits, now) {
     return Math.max(0, rateLimitAfter, banAfter);
   };
 
+  EmailRecord.prototype.isOverBadLogins = function() {
+    this.trimBadLogins(now());
+    return this.lf.length > limits.maxBadLoginsPerEmail;
+  };
+
+  EmailRecord.prototype.addBadLogin = function() {
+    this.trimBadLogins(now());
+    this.lf.push(now());
+  };
+
+  EmailRecord.prototype.trimBadLogins = function(now) {
+    if (this.lf.length === 0) {
+      return;
+    }
+    // lf is naturally ordered from oldest to newest
+    // and we only need to keep up to limits.maxBadLoginsPerEmail + 1
```

Comment:
since our memcache ops aren't atomic I'd rather do a sort here and not have to think about order even if it's impossible for it to be out of order. this list is small